src/main/java/org/springframework/data/elasticsearch/core/MappingBuilder.java

```diff
@@ -360,6 +364,9 @@ private void addFieldMappingParameters(XContentBuilder builder, Object annotatio
 			searchAnalyzer = fieldAnnotation.searchAnalyzer();
 			normalizer = fieldAnnotation.normalizer();
 			copyTo = fieldAnnotation.copyTo();
+			if (!FieldType.Text.equals(type) && !FieldType.Nested.equals(type)) {
```

Comment:
If we throw an error in case of doc_values=false, then we should throw an error if the user uses doc_values=true case as well. But in case Field.docValues() with the value true, it is difficult to differentiate it is a default value or user-specified value.

Reply:
in what case would _doc_values == true_ be an error? That's the default value and we shouldn't write it to the mapping anyway, only in the case that it's set to f alse

Reply:
Means for the Text type field if the user specifies doc_value=false. We should throw an error. If the user specifies doc_value=true, we should ignore. Is it my understanding correct?

Reply:
yes. a value of true is not written to the mapping anyway. From the docs: _All fields which support doc values have them enabled by default._ So setting to _true_ is just the default behaviour.

Reply:
is this creates misunderstanding? May be user thinks doc_value=true supported by the system not the false value for the nonsupported field types like Text or Nested.

Reply:
on nonsupported fields there may be no doc_value at all, neither true or false

Reply:
What my question is for the Text type field if the user specifies the _doc_values=true_, if we don't throw an error. User may think it is valid. I think we should throw an error in case of _doc_values=true_ as well. What is your opinion about this?

Reply:
what about fields of type text with index=false? I think we need to try the different constellations to find the valid combinations

Reply:
ok, I checked, and _doc_value_ on a text field with _index=flse_ is not allowed. But we need tests for the different other cases to make sure that the _doc_values_ is written in the mapping (you can see in the tests that use the `FieldnameEntity` which special cases exists